Zilon.Core/Zilon.Core.Tests/Tactics/ActUsageHandlerSelectorTests.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+
+using FluentAssertions;
+
+using Moq;
+
+using NUnit.Framework;
+
+using Zilon.Core.Graphs;
+using Zilon.Core.Persons;
+
+namespace Zilon.Core.Tactics.Tests
+{
+    [TestFixture]
+    [Parallelizable(ParallelScope.All)]
+    public class ActUsageHandlerSelectorTests
+    {
+        /// <summary>
+        /// Тест проверяет, что селектор корректно возвращает обработчик,
+        /// если целевой объект напрямую реализует тестовый интерфейс.
+        /// </summary>
+        [Test]
+        public void GetHandler_TargetDirectlyImplementTargetInterfact_ReturnsHandler()
+        {
+            // ARRANGE
+
+            var testHandlerMock = new Mock<IActUsageHandler>();
+            testHandlerMock.SetupGet(x => x.TargetType).Returns(typeof(ITest1));
+            var testHandler = testHandlerMock.Object;
+
+            var selector = new ActUsageHandlerSelector(new IActUsageHandler[] { testHandler });
+
+            var targetObject = new Test1();
+
+            // ACT
+
+            var factHandler = selector.GetHandler(targetObject);
+
+            // ASSERT
+
+            factHandler.Should().Be(testHandler);
+        }
+
+        /// <summary>
+        /// Тест проверяет, что селектор корректно возвращает обработчик,
+        /// если целевой объект унаследован от базового класса, который реализует целевой интерфейс.
+        /// </summary>
+        [Test]
+        public void GetHandler_TargetInheritedByBase_ReturnsHandler()
+        {
+            // ARRANGE
+
+            var testHandlerMock = new Mock<IActUsageHandler>();
+            testHandlerMock.SetupGet(x => x.TargetType).Returns(typeof(ITest1));
+            var testHandler = testHandlerMock.Object;
+
+            var selector = new ActUsageHandlerSelector(new IActUsageHandler[] { testHandler });
+
+            var targetObject = new Test2();
+
+            // ACT
+
+            var factHandler = selector.GetHandler(targetObject);
+
+            // ASSERT
+
+            factHandler.Should().Be(testHandler);
+        }
+
+        /// <summary>
+        /// Тест проверяет, что селектор возвращает null (НЕ во),
+        /// если тип целевого объекта не приводится к целевому типу обработчика.
+        /// </summary>
+        [Test]
+        public void GetHandler_IsNotHandleTarget_ThrowsExceptionHandlerNotFound()
+        {
+            // ARRANGE
+
+            var testHandlerMock = new Mock<IActUsageHandler>();
+            testHandlerMock.SetupGet(x => x.TargetType).Returns(typeof(ITest1));
+            var testHandler = testHandlerMock.Object;
+
+            var selector = new ActUsageHandlerSelector(new IActUsageHandler[] { testHandler });
+
+            var targetObject = new NotHandleTarget();
+
+            // ACT
+            Action act = () =>
+            {
+                var factHandler = selector.GetHandler(targetObject);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'factHandler'.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5403398)